### ISSUE TITLE
Is it recommended to use unblock for CPU intensive operations?

### ISSUE THREAD

**Issue by ItsSunnyMonster** [smol-rs/blocking#64]
e.g. iterating through a long vector and doing operations on each item. 

The docs says use this for blocking IO, but does not mention CPU bound functions. I saw some discussions about tokio's `spawn_blocking` but I'm not sure if the same will apply here.

**Comment by notgull** [comment]
Generally yes. `async` tasks are designed around the fact that any given task will run for a few nano/microseconds before yielding. So if you have a very CPU intensive operation you should unblock it.

However! You can avoid blocking other tasks by yielding in the middle of a long vector. For example, if you are doing this:

```rust
for foo in bar {
    do_something(foo);
}
```

You can do this instead, yielding every 1024 items and letting another task run if one is queued.

```rust
for (i, foo) in bar.enumerate() {
    do_something(foo);
    if i % 1024 == 0 {
        futures_lite::futures::yield_now().await;
    }
}
```

**Comment by ItsSunnyMonster** [comment]
Wow, I had no idea `yield_now` existed. Thank you!
